llvm/lib/Target/AMDGPU/VOPInstructions.td

```diff
@@ -819,9 +825,7 @@ class VOP3P_DPPe_Common<bits<7> op, VOPProfile P> : VOP3P_DPPe_Common_Base<op, P
 
 class VOP_DPP_Pseudo <string OpName, VOPProfile P, list<dag> pattern=[],
   dag Ins = P.InsDPP, string asmOps = P.AsmDPP> :
-  InstSI <P.OutsDPP, Ins, OpName#asmOps, pattern>,
-  VOP <OpName>,
-  SIMCInstr <OpName#"_dpp", SIEncodingFamily.NONE> {
+  VOP_Pseudo<OpName, "_dpp", P, P.OutsDPP, Ins, asmOps, pattern> {
```

Comment:
Why this change - is it just a cleanup?

Reply:
This change allows dpp instructions to be included in the searchable table by inheriting the IsInvalidSingleUse* bits from VOP_Pseudo.

Reply:
I see. I think this just allows you to use VOP_Pseudo as a common superclass for all VOP instructions, DPP or not, when you define the searchable table. Perhaps you could use an even more common class like InstSI - I'm not sure if that would make the generated table bigger.

@Sisyph @kosarev any thoughts on this part of the patch?

Reply:
Looks reasonable to me and a refinement as well. I second Jay's words on whether it makes sense to limit the flags to `VOP_Pseudo`s and then have explicit `SIInstrInfo::isVAL()` checks, though.

Reply:
As a followup I guess we could remove a lot of the definitions below like `let isPseudo = 1` that can now be inherited from `VOP_Pseudo`.